parsers/PA.py

```diff
@@ -97,33 +110,67 @@
     "Tropitérmica 3": "oil",  # [6]:162[7] spelled "Tropitermica" in both
 }
 
+PRODUCTION_TYPE_TO_PRODUCTION_MODE = {
+    "Hídrica": "hydro",
+    "Eólica": "wind",
+    "Solar": "solar",
+    "Biogás": "biomass",
+    "Térmica": "unknown",
+}
+
+_SPANISH_CALENDAR = {
+    "enero": "01",
+    "febrero": "02",
+    "marzo": "03",
+    "abril": "04",
+    "mayo": "05",
+    "junio": "06",
+    "julio": "07",
+    "agosto": "08",
+    "septiembre": "09",
+    "octubre": "10",
+    "noviembre": "11",
+    "diciembre": "12",
+}
+
+
+def _localise_spanish_date(date: str) -> str:
+    """Localises a date containing full name (lowercase) spanish months, replacing them with zero-padded decimal numbers.
+
+    This avoids having to mess up the global locale to be able to parse the date.
+    """
+    return re.sub(
+        "|".join(_SPANISH_CALENDAR.keys()),
+        lambda m: _SPANISH_CALENDAR[m.group(0)],
+        date,
+    )
```

Comment:
note: the production API returns 'spanish_dates' like `'31-marzo-2024 5:26:36'`

`arrow` had a neat one-liner to parse them:
`date = arrow.get(spanish_date, "DD-MMMM-YYYY H:mm:ss", locale="es", tzinfo="America/Panama")`

but unfortunately `datetime.strptime()` doesn't have a similar way to provide a locale and we would be forced to set it globally for it to work - [which is problematic](https://stackoverflow.com/a/32785195)

This function "translates" the month to its corresponding number so that then it can be parsed in a locale-independent way